SCons/Script/Main.py

```diff
@@ -1281,16 +1271,19 @@ def tmtrace_cleanup(tfile):
     # As of 3.7, python removed support for threadless platforms.
     # See https://www.python.org/dev/peps/pep-0011/
     is_37_or_later = sys.version_info >= (3, 7)
-    python_has_threads = sysconfig.get_config_var('WITH_THREAD') or is_pypy or is_37_or_later
+    # python_has_threads = sysconfig.get_config_var('WITH_THREAD') or is_pypy or is_37_or_later
+
+    # As of python 3.4 threading has a dummy_threading module for use when there is no threading
+    # it's get_ident() will allways return -1, while real threading modules get_ident() will
+    # always return a positive integer
+    python_has_threads = threading.get_ident() != -1
     # to check if python configured with threads.
     global num_jobs
     num_jobs = options.num_jobs
     jobs = SCons.Job.Jobs(num_jobs, taskmaster)
     if num_jobs > 1:
         msg = None
-        if sys.platform == 'win32':
-            msg = fetch_win32_parallel_msg()
-        elif jobs.num_jobs == 1 or not python_has_threads:
+        if jobs.num_jobs == 1 or not python_has_threads:
```

Comment:
This check doesn't make sense... we're already inside a block which is `if num_jobs > 1` so the first part can't be true in any case, and if it were, it's not when we'd want to generate the warning.  Further, if we're actually ignoring `-j`, shouldn't we set `num_jobs` down to one to make that true?

Reply:
num_jobs comes from -j, but jobs.num_jobs comes from SCons.Job.Jobs which can be one, if ```                self.job = Parallel(taskmaster, num, stack_size)
``` Throws an exception.